homeassistant/components/switch/rest.py

```diff
@@ -36,26 +37,38 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
         _LOGGER.error("No route to resource/endpoint: %s", resource)
         return False
 
+    cfg_headers = config.get('headers')
+
+    if cfg_headers is None:
+        headers = DEFAULT_HEADERS
+    else:
+        headers = {}
+        for header in cfg_headers:
```

Comment:
I am very confused as to what you are doing here. Headers will already be a dictionary so no need for this "reformatting"